### PR TITLE
feature/networkPeering: Added Azure 'Cross Subscription Network Peerings' Plugin and test cases

### PR DESCRIPTION
**Pull request by ali-imran7**

### CODE REVIEW

File: plugins/azure/virtualnetworks/virtualNetworkPeering.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure/');
+
+module.exports = {
+    title: 'Cross Subscription Network Peering',
```
Comment (AkhtarAmir) [inline plugins/azure/virtualnetworks/virtualNetworkPeering.js:5]:
```suggestion
    title: 'Virtual Network Peering',
```

File: plugins/azure/virtualnetworks/virtualNetworkPeering.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    category: 'Virtual Networks',
+    description: 'Ensures that Virtual Network is not connected with a virtual network in whitelisted subscription.',
```
Comment (AkhtarAmir) [inline plugins/azure/virtualnetworks/virtualNetworkPeering.js:7]:
```suggestion
    description: 'Ensures that Virtual Network has peering connection only with a virtual network in whitelisted subscription.',
```

File: plugins/azure/virtualnetworks/virtualNetworkPeering.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    more_info: 'To meet your organization\'s security compliance requirements.',
```
Comment (AkhtarAmir) [inline plugins/azure/virtualnetworks/virtualNetworkPeering.js:8]:
```suggestion
    more_info: 'Virtual networks should only have peering connections with whitelisted virtual networks to meet your organization\'s security compliance requirements.',
```

File: plugins/azure/virtualnetworks/virtualNetworkPeering.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    link: 'https://docs.microsoft.com/en-us/azure/virtual-network/virtual-network-peering-overview',
+    recommended_action: 'Delete Peering Connection with the subscription which are not whitelisted',
+    apis: ['virtualNetworks:listAll', 'virtualNetworkPeerings:list'],
+    settings: {
+        peering_denied_subscriptions: {
+            name: 'Subcriptions, Virtual Network Peering Denied for',
+            description: 'Subscription Ids, for which Virtual Network Peering in not allowed with current subscription networks',
+            regex: '/^([a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12},?)+$/',
+            default: ''
+        }
```
Comment (AkhtarAmir) [inline plugins/azure/virtualnetworks/virtualNetworkPeering.js:18]:
```suggestion
        whitelisted_peering_subscriptions: {
            name: 'Whitelisted Peering Subscriptions',
            description: 'Subscription Ids for remote virtual networks which should be allowed for peering',
            regex: '/^([a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12},?)+$/',
            default: ''
        }
```

File: plugins/azure/virtualnetworks/virtualNetworkPeering.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                    return scb();
+                }
+    
+                if (!virtualNetworkPeerings.data.length) {
+                    helpers.addResult(results, 0, 'No existing Virtual Network Peerings found', location);
+                    return scb();
+                }
+
+                virtualNetworkPeerings.data.forEach(peering => {
+                    let deniedSubscriptions = config.subscriptions.split(',');
```
Comment (AkhtarAmir) [inline plugins/azure/virtualnetworks/virtualNetworkPeering.js:61]:
```suggestion
```

File: plugins/azure/virtualnetworks/virtualNetworkPeering.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                    let subscriptionId = '';
+                    if (peering.remoteVirtualNetwork && peering.remoteVirtualNetwork.id) {
+                        subscriptionId = peering.remoteVirtualNetwork.id.split('/')[2];
+                    }
+
+                    if (deniedSubscriptions.length && deniedSubscriptions.includes(subscriptionId)) {
+                        helpers.addResult(results, 2, 'Virtual network is not connected with a virtual network in whitelisted subscription', location, virtualNetwork.id);
```
Comment (AkhtarAmir) [inline plugins/azure/virtualnetworks/virtualNetworkPeering.js:68]:
```suggestion
                        helpers.addResult(results, 2, `Vitual network has peering with these unknown subscriptions: ${unknownSubscriptions.join(', ')}`, location, virtualNetwork.id);
```

File: plugins/azure/virtualnetworks/virtualNetworkPeering.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+                    } else {
+                        helpers.addResult(results, 0, 'Virtual network is connected with a virtual network in whitelisted subscription', location, virtualNetwork.id);
+                    }
+                });
+
+                scb();
+            }, function(){
+                rcb(null, results, source);
```
Comment (AkhtarAmir) [inline plugins/azure/virtualnetworks/virtualNetworkPeering.js:76]:
```suggestion
                rcb();
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Review by giorod3 — CHANGES_REQUESTED** [review]
change to opt-in

**Review by giorod3 — APPROVED** [review]

**Pull request merged**